ethereumetl/bifrost.py

```diff
@@ -0,0 +1,102 @@
+# MIT License
+#
+# Copyright (c) 2019 Nirmal AK, nirmal@merklescience.com
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import requests
+from datetime import datetime
+from urllib3.util import Retry
+from requests.adapters import HTTPAdapter
+
+BIFRSOT_AUTH = os.getenv("BIFROST_AUTH", "")
+
+
+class BifrostRequestException(Exception):
+    pass
+
+
+def _make_request(
+        from_currency_code: str,
+        to_currency_code: str,
+        execution_date: str,
+        access_token: str,
+
+) -> requests.Response:
+    base_url = f"https://dev.bifrost.palantree.com/coin-price-service/historical-daily-price/"
+    retries = Retry(backoff_factor=5, total=5)
+    adapter = HTTPAdapter(max_retries=retries)
+    session = requests.Session()
+    session.mount('https://', adapter)
+    session.mount('http://', adapter)
+    retries.BACKOFF_MAX = 600
```

Comment:
I forget what is the inital sleep ?
2 secs?

Reply:
5 sec